core/adapters/base.ts

```diff
@@ -22,18 +23,50 @@ export abstract class Adapter {
 
   protected insertInto(target: dataform.ITarget, columns: string[], query: string) {
     return `
-      insert into ${this.resolveTarget(target)}
-      (${columns.join(",")})
-      select ${columns.join(",")}
-      from (${query}) as insertions`;
+insert into ${this.resolveTarget(target)}
+(${columns.join(",")})
+select ${columns.join(",")}
+from (${query}) as insertions`;
   }
 
   protected oppositeTableType(type: string) {
     return this.baseTableType(type) === "table" ? "view" : "table";
   }
 
   protected where(query: string, where: string) {
-    return `select * from (${query}) as subquery
-        where ${where || "true"}`;
+    return `
+  select * from (${query}) as subquery
+    where ${where || "true"}`;
+  }
+
+  protected refreshIncrementCondition(
+    runConfig: dataform.IRunConfig,
+    tableMetadata: dataform.ITableMetadata
+  ) {
+    return runConfig.fullRefresh || !tableMetadata || tableMetadata.type === "view";
+  }
+
+  protected addPreOps(
```

Comment:
IMO it's weird that this causes state changes. i think it should return the list of preops which should be added, as a list of `Task`s, and should be called `preOps`

Reply:
Also added `||` check to incrementalPreOps as proto allows a null value